compiler/qsc_rca/src/overrider.rs

```diff
@@ -0,0 +1,469 @@
+        self.package_store_compute_properties
+            .insert_spec((package_id, spec_id).into(), application_generator_set);
+    }
+
+    fn set_current_application_generator_set(&mut self, value: ApplicationGeneratorSet) {
+        assert!(self.current_application_generator_set.is_none());
+        self.current_application_generator_set = Some(value);
+    }
+}
+
+impl<'a> Visitor<'a> for Overrider<'a> {
+    fn get_block(&self, id: BlockId) -> &'a Block {
+        let package_id = self.get_current_package();
+        self.package_store.get_block((package_id, id).into())
+    }
+
+    fn get_expr(&self, id: ExprId) -> &'a Expr {
+        let package_id = self.get_current_package();
+        self.package_store.get_expr((package_id, id).into())
+    }
+
+    fn get_pat(&self, id: PatId) -> &'a Pat {
+        let package_id = self.get_current_package();
+        self.package_store.get_pat((package_id, id).into())
+    }
+
+    fn get_stmt(&self, id: StmtId) -> &'a Stmt {
+        let package_id = self.get_current_package();
+        self.package_store.get_stmt((package_id, id).into())
+    }
+
+    fn visit_block(&mut self, id: BlockId) {
+        walk_block(self, id);
+        let package_id = self.get_current_package();
+        let block = self.get_block(id);
+        let application_generator_set = adapt_application_generator_set_to_type(
+            self.get_current_application_generator_set(),
+            &block.ty,
+        );
+        self.package_store_compute_properties
+            .insert_block((package_id, id).into(), application_generator_set);
+    }
+
+    fn visit_expr(&mut self, id: ExprId) {
+        walk_expr(self, id);
+        let package_id = self.get_current_package();
+        let expr = self.get_expr(id);
+        let application_generator_set = adapt_application_generator_set_to_type(
+            self.get_current_application_generator_set(),
+            &expr.ty,
+        );
+        self.package_store_compute_properties
+            .insert_expr((package_id, id).into(), application_generator_set);
+    }
+
+    fn visit_package(&mut self, package: &'a Package) {
+        // Go through each namespace, identifying the callables for which we have overrides.
+        let namespaces = package
+            .items
+            .iter()
+            .filter_map(|(_, item)| match &item.kind {
+                ItemKind::Namespace(ident, items) => Some((ident.name.to_string(), items)),
+                _ => None,
+            });
+        for (namespace_ident, namespace_items) in namespaces {
+            let callables = namespace_items
+                .iter()
+                .filter_map(|item_id| {
+                    let item = self.get_item(*item_id);
+                    match &item.kind {
+                        ItemKind::Callable(decl) => Some((item.id, decl.name.name.to_string())),
+                        _ => None,
+                    }
+                })
+                .collect_vec();
```

Comment:
Again here, this is not needed:

```suggestion
                .collect::<Vec<_>>();
```